numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -1828,6 +1830,61 @@ call_npy_@name@@suffix@(PyObject *NPY_UNUSED(self), PyObject *args)
 
 /**end repeat**/
 
+/*
+ * For development/testing purposes, it's conventient to have access to the
+ * system printf for floats. This is a very simple printf interface.
+ */
+PyObject *
+PrintFloat_Printf_g(PyObject *obj, int precision)
+{
+    char str[1024];
+
+    if (PyArray_IsScalar(obj, Half)) {
+        npy_half x = ((PyHalfScalarObject *)obj)->obval;
+        PyOS_snprintf(str, sizeof(str), "%.*g", precision,
+                      npy_half_to_double(x));
+    }
+    else if (PyArray_IsScalar(obj, Float)) {
+        npy_float x = ((PyFloatScalarObject *)obj)->obval;
+        PyOS_snprintf(str, sizeof(str), "%.*g", precision, x);
+    }
+    else if (PyArray_IsScalar(obj, Double)) {
+        npy_double x = ((PyDoubleScalarObject *)obj)->obval;
+        PyOS_snprintf(str, sizeof(str), "%.*g", precision, x);
```

Comment:
Would be nice to use `lg` here, but that's not supported in C90 :(

Reply:
Might be nice to add a comment to that effect, actually

Reply:
I see some comments online that it `lg` is more consistent with scanf, is that why?

Reply:
Yeah, that's why. `va_list` doesn't support `float` which is why it doesn't actually matter for printf (but it does support `float *`, which is why it matters for scanf)